source/extensions/clusters/redis/redis_cluster_lb.cc

```diff
@@ -0,0 +1,83 @@
+#include "redis_cluster_lb.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Redis {
+
+// RedisClusterLoadBalancerFactory
+bool RedisClusterLoadBalancerFactory::onClusterSlotUpdate(
+    const std::vector<Envoy::Extensions::Clusters::Redis::ClusterSlot>& slots,
+    Envoy::Upstream::HostMap all_hosts) {
+
+  absl::WriterMutexLock lock(&mutex_);
```

Comment:
You only need to hold the writer lock write before you update the member shared pointer value.

Reply:
that's what I had before, but clang treat it as an error.  Is there a way to disable that check here? Alternatively I do a reader lock first, then a writer lock when we actually do the update?